### ISSUE TITLE
Possible memory leak on 1.1.2

### ISSUE THREAD

**Issue by Tri125** [rdvojmoc/DinkToPdf#98]
In our netcore 2.2 MVC project we're using DinkToPdf to generate a bunch of PDF from a cshtml template.

We're registering a SynchronizedConverter as a singleton to inject it into our services:

```csharp
services.AddSingleton(typeof(IConverter), new SynchronizedConverter(new PdfTools()));
```

and our usage looks like that:

```csharp
foreach (var questionnaireInformation in zipInformation.QuestionnaireInformations)
{
     var s = await this.viewRenderService.RenderViewAsync(controller, "~/Views/myTemplate.cshtml", model);

     var pdf = this.pdfConverter.Convert(new HtmlToPdfDocument
     {
         GlobalSettings = { ColorMode = ColorMode.Color, Orientation = Orientation.Portrait, PaperSize = PaperKind.A4 },
         Objects =
         {
             new ObjectSettings
             {
	        UseLocalLinks = true,
                PagesCount = true,
                HtmlContent = s,
                WebSettings = { DefaultEncoding = "utf-8"},
             }
         }
    });

    // Write pdf to a stream, etc.
}
```

We were already using DinkToPdf for individual pdf but when we started to generate hundred of them together like in the example above, we noticed that the memory would go through the roof and it would never be garbage collected.  A quick memory snapshot tells me that the count of some objects related to DinkToPdf are increasing.

**Comment by WebDev-Guy** [comment]
@Tri125 I can confirm your assumptions. We are experiencing the same issue.

**Comment by SeanLeitzinger** [comment]
@Tri125 @drewgierach Did you guys ever confirm the memory leak issue?

**Comment by Tri125** [comment]
@SeanLeitzinger 

Well I'm pretty confident that there's a leak because of what I described above. It's up to the dev team to analyse our report and see what's going on.

**Comment by SeanLeitzinger** [comment]
I don't think this repository is maintained anymore. I'm in the process of building out the Html To Image functionality and will probably see if I can duplicate the leak on the PDF side as well.
